src/app/shared/auth/auth-info.spec.ts

```diff
@@ -1,72 +1,86 @@
 import { SECONDS } from '../helpers/duration';
-import { clearTokenFromStorage, tokenAuthFromServiceResp, tokenAuthFromStorage } from './auth-info';
+import {
+  clearForcedTokenFromStorage,
+  clearTokenFromStorage,
+  forcedTokenAuthFromStorage,
+  hasForcedToken,
+  setForcedTokenInStorage,
+  tokenAuthFromServiceResp,
+  tokenAuthFromStorage
+} from './auth-info';
 
 describe('auth-info', () => {
 
+  const expectedToken = 'atoken';
   const expiresInSec = 60;
   const expiresInMS = expiresInSec*SECONDS;
   const precision = -3; // the 3 lower digits may be different = 1s precison
 
   beforeEach(function() {
     clearTokenFromStorage();
+    clearForcedTokenFromStorage();
   });
 
   it('should save & load successfully a token', () => {
     const now = Date.now();
 
-    const auth = tokenAuthFromServiceResp('atemptok', expiresInSec);
+    const auth = tokenAuthFromServiceResp(expectedToken, expiresInSec);
     const loaded = tokenAuthFromStorage();
 
-    expect(auth.accessToken).toEqual('atemptok');
+    expect(auth.accessToken).toEqual(expectedToken);
     expect(auth.creation.getTime()-now).toBeCloseTo(0,precision);
     expect(auth.expiration.getTime()-now).toBeCloseTo(expiresInMS,precision);
 
-    expect(loaded?.accessToken).toEqual('atemptok');
-    if (loaded) {
-      expect(loaded.creation.getTime()).toBeCloseTo(now,precision);
-      expect(loaded.expiration.getTime()).toBeCloseTo(now+expiresInMS,precision);
-    }
+    expect(loaded.accessToken).toEqual(expectedToken);
+    expect(loaded.creation.getTime()).toBeCloseTo(now,precision);
+    expect(loaded.expiration.getTime()).toBeCloseTo(now+expiresInMS,precision);
   });
 
   it('should fail when loading a token with no token', () => {
-    tokenAuthFromServiceResp('atemptok', expiresInSec);
+    tokenAuthFromServiceResp(expectedToken, expiresInSec);
     sessionStorage.removeItem('access_token');
-    const loaded = tokenAuthFromStorage();
-    expect(loaded).toBeUndefined();
+    expect(tokenAuthFromStorage).toThrowError();
   });
 
   it('should fail when loading a token with no expiration', () => {
-    tokenAuthFromServiceResp('atemptok', expiresInSec);
+    tokenAuthFromServiceResp(expectedToken, expiresInSec);
     sessionStorage.removeItem('access_token_exp');
-    const loaded = tokenAuthFromStorage();
-    expect(loaded).toBeUndefined();
+    expect(tokenAuthFromStorage).toThrowError();
   });
 
   it('should fail (and clear) when loading a expired token', () => {
-    tokenAuthFromServiceResp('atemptok', -1 /* expired 1sec ago */);
-    const loaded = tokenAuthFromStorage();
-    expect(loaded).toBeUndefined();
+    tokenAuthFromServiceResp(expectedToken, -1 /* expired 1sec ago */);
+    expect(tokenAuthFromStorage).toThrowError();
     expect(sessionStorage.getItem('access_token')).toBeNull();
     expect(sessionStorage.getItem('access_token_exp')).toBeNull();
```

Comment:
I didn't spot this one before, I think it's implementation detail and it makes the test more fragile. What do you think ?
I spotted others in the PR in case you think it should be changed.